docs/aws.md

```diff
@@ -1,39 +1,213 @@
+
+* [Installation Guide](http://kubernetes.io/docs/user-guide/prereqs/)
+
+
+```bash
+wget -O https://github.com/kubernetes/kubernetes/releases/download/v1.4.6/kubernetes.tar.gz
+sudo cp kubernetes/platforms/darwin/amd64/kubectl /usr/local/bin/kubectl
 ```
 
-If you have problems, please set `--v=8` and open an issue, and ping justinsb on slack!
+## Setup your environment
+
+### Setting up a kops IAM user
+
+
+In this example we will be using a dedicated IAM user to use with kops. This user will need basic API security credentials in order to use kops. Create the user and credentials using the AWS console. [More information](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html).
+
+Kubernetes kops uses the official AWS Go SDK, so all we need to do here is set up your system to use the official AWS supported methods of registering security credentials defined [here](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials). Here is an example using the aws command line tool to set up your security credentials.
+
+```bash
+brew update && brew install awscli
+aws configure
+aws iam list-users
+```
+
+We should now be able to pull a list of IAM users from the API, verifying that our credentials are working as expected.
+
+## Configure DNS
+
+We will now need to set up DNS for cluster, find one of the scenarios below (A,B,C) that match your situation.
+
+### (A) Setting up DNS for your cluster, with amazon as your registrar
+
+If you bought your domain with AWS, then you should already have a hosted zone in Route53.
+
+If you plan on using your base domain, then no more work is needed. If you plan on using a subdomain to build your clusters on you will need to create a 2nd hosted zone in Route53.
+
+```bash
+ID=$(uuidgen) && aws route53 create-hosted-zone --name subdomain.kubernetes.com --caller-reference $ID
+```
+
+### (B) Setting up DNS for your cluster, with another registrar.
+
+If you bought your domain elsewhere, and would like to dedicate the entire domain to AWS you should follow the guide [here](http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/domain-transfer-to-route-53.html)
+
+### (C) Setting up a subdomain for clusters, with another registrar while keeping your top level domain the same
+
+If you bought your domain elsewhere, but **only want to use a subdomain in AWS Route53** you must modify your registrar's NS (NameServer) records. See the example below.
+
+Here we will be creating a hosted zone in AWS Route53, and migrating the subdomain's NS records to your other registrar.
+
+  - Create the subdomain
+
+```bash
+ID=$(uuidgen) && aws route53 create-hosted-zone --name subdomain.kubernetes.com --caller-reference $ID
+```
+
+  - Note your hosted zone ID
+
+```bash
+aws route53 list-hosted-zones | jq '.HostedZones[] | select(.Name=="subdomain.kubernetes.com") | .Id'
```

Comment:
we nee jq for that example ... mention install https://github.com/stedolan/jq/wiki/Installation